README.md

```diff
@@ -1,634 +1,1828 @@
+  -- UpdateLenderAddress spending redeemer.
+  = ObserveAddressUpdate
+  -- | Register the script for staking execution. This can only be used with a certificate
+  -- execution.
+  | RegisterAddressUpdateObserverScript
+```
 
+##### Loan Spending Smart Contract Redeemers
+```haskell
+data LoanRedeemer
+  -- | Close or update an Ask UTxO.
+  = CloseOrUpdateAsk
+  -- | Close or update an Offer UTxO.
+  | CloseOrUpdateOffer
+  -- | Convert an Ask UTxO and an Offer UTxO into an Active UTxO.
+  | AcceptOffer
+  -- | Make a payment on a loan. The amount is the size of the payment made for this loan.
+  | MakePayment { paymentAmount :: Integer }
+  -- | Apply interest to this loan N times and deposit the specified amount of ada if needed for the
+  -- minUTxOValue.
+  | ApplyInterest { depositIncrease :: Integer, numberOfTimes :: Integer }
+  -- | Claim collateral for an expired loan using the Key NFT.
+  | SpendWithKeyNFT
+  -- | Update the address where loan payments must go. Optionally deposit additional ada if needed
+  -- for the minUTxOValue.
+  | UpdateLenderAddress { newAddress :: Address, depositIncrease :: Integer }
+  -- | Clean up remaining beacons from Finished Active UTxOs and/or claim Lost Active UTxOs.
+  | Unlock
+  deriving (Generic,Show)
+```
 
-``` Haskell
-data LoanDatum
-  -- | The datum for the ask phase.
-  = AskDatum 
-      { loanBeaconSym :: CurrencySymbol -- ^ Policy Id of the beacon minting policy.
-      , borrowerId :: TokenName -- ^ The staking pubkey hash for the borrower.
-      , loanAsset :: (CurrencySymbol,TokenName) -- ^ The asset to be loaned out.
-      , loanPrinciple :: Integer -- ^ The amount of the loan asset to be loaned out.
-      , loanTerm :: POSIXTime -- ^ The length of time where the loan will be active.
-      , collateral :: [(CurrencySymbol,TokenName)] 
-          -- ^ All assets the borrower is willing to use as collateral.
+##### Loan Spending Smart Contract Datums
+A DApp UTxO's datum can either be an `AskDatum`, an `OfferDatum`, or an `ActiveDatum`. `POSIXTime`
+is always in milliseconds.
+
+```haskell
+-- | A type synonym for an asset's full name.
+type Asset = (CurrencySymbol,TokenName)
+
+-- | The penalty to apply whenever the minimum payment is not met in a given compounding period.
+data Penalty
+  = NoPenalty
+  | FixedFee Integer
+  | PercentFee Rational
+
+data LoanDatum 
+  = AskDatum
+      -- | The policy id for the negotiation beacon script.
+      { negotiationBeaconId :: CurrencySymbol
+      -- | The policy id for the active beacon script.
+      , activeBeaconId :: CurrencySymbol
+      -- | The borrower's staking credential hash as a token name.
+      , borrowerId :: TokenName
+      -- | The asset to be loaned out.
+      , loanAsset :: Asset
+      -- | The token name for the loan asset beacon.
+      , assetBeacon :: TokenName
+      -- | The size of the loan.
+      , loanPrinciple :: Integer
```

Comment:
```suggestion
      -- | The size of the loan.
      , loanPrincipal :: Integer
```
The term "Principle" (just in this machine readable term as I can see) are spelled wrong throughout the document: https://www.investopedia.com/terms/p/principal.asp

Reply:
This entire time I thought it was "loan principle" :upside_down_face:

Reply:
`loanPrinciple :: String` == Why you are borrowing the money. 🤣